Sources/TransifexNative/Logging.swift

```diff
@@ -8,39 +8,76 @@
 @objc
-public enum TXLoggingLevel : Int {
-    /// Info messages
-    case info
-    /// Warning messages
-    case warning
-    /// Error messages
-    case error
-    /// Verbose messages
-    case verbose
+public protocol TXLogHandler {
+    /// Logs an info message.
+    /// - Parameter message: The info message
+    func info(_ message: String)
+    
+    /// Logs a warning message.
+    ///
+    /// - Parameter message: The warning message
+    func warning(_ message: String)
+    
+    /// Logs an error message.
+    ///
+    /// - Parameter message: The error message
+    func error(_ message: String)
+    
+    /// Logs a verbose message.
+    ///
+    /// - Parameter message: The verbose message
+    func verbose(_ message: String)
 }
 
-/// Protocol that can be used to control all logging that occurs in the SDK
-@objc
-public protocol TXLogger {
-    /// Logs a message with a certain log level.
+/// Class responsible for control logging in the SDK, allowing external log handlers
+/// to be updated by SDK clients via the `TXLogHandler` protocol.
+public final class TXLogger: NSObject {
+    fileprivate var handler: TXLogHandler?
+    
+    fileprivate init(handler: TXLogHandler) {
+        self.handler = handler
+    }
+    
+    /// Updates the log handler of the SDK with an external class that conforms to
+    /// the `TXLogHandler` protocol.
     ///
-    /// - Parameters:
-    ///   - message: The message to be logged.
-    ///   - level: The message's log level.
-    func log(message: String, level: TXLoggingLevel)
+    /// - Parameter handler: The new log handler to be used
+    @objc
+    public static func setHandler(handler: TXLogHandler) {
+        Logger.handler = handler
+    }
+    
+    // Pragma: Delegate all log calls to the internal log handler.
+    
+    func info(_ message: String) {
+        handler?.info(message)
+    }
+    
+    func warning(_ message: String) {
+        handler?.warning(message)
+    }
+    func error(_ message: String) {
+        handler?.error(message)
+    }
+    func verbose(_ message: String) {
+        handler?.verbose(message)
+    }
 }
 
-/// The default logger used by the SDK if no other logging was provided by the developer. Only error logs are
-/// written to the console using the `print()` function.
-class ErrorLogger : TXLogger {
-    func log(message: String, level: TXLoggingLevel) {
-        guard level == .error else {
-            return
-        }
-        
+/// Default log handler to be used if client doesn't provide a new one.
+///
+/// Only the error log messages are printed to the console, every other log
+/// message method of `TXLogHandler` protocol is a no-op.
+fileprivate class StandardLogHandler : TXLogHandler {
+    func info(_ message: String) { }
+    func warning(_ message: String) { }
+    func verbose(_ message: String) { }
+    func error(_ message: String) {
         print(message)
     }
 }
```

Comment:
The way I had this in mind is to make it public, as a fast way for the clients to define a different log level. e.g.

```swift
public class StandardLogHandler : TXLogHandler {

    private var logLevel: TXLogLevel;

    init(_ logLevel: TXLogLevel) {
        self.logLevel = logLevel
    }
    
    func verbose(_ message: String) { log(message, .verbose) }
    func info(_ message: String) { log(message, .info) }
    func warning(_ message: String) { log(message, .warning) }
    func error(_ message: String) { log(message, .error) }

    private func log(_ message: String, logLevel: TXLogLevel) {
        guard logLevel >= self.logLevel else {
            return
        }
        print(message)
    }

}
```

and in the client:

```swift
TXLogger.setHandler(StandardLogHandler(.info))
```

This way, devs won't have to create their only handler class simply in order to set a new log level.

Reply:
Just pushed the commit with the updated implementation.